pyproject.toml

```diff
@@ -42,6 +42,16 @@ rich = "^13.0"
 # Extras dependencies
 xarray = { version = ">2024.0.0", optional = true }
 numpy = { version = "^2.0.0", optional = true }
+pytest = { version = "*", optional = true }
+pytest-benchmark = { version = "*", optional = true }
+pytest-randomly = { version = "*", optional = true }
+pytest-cov = { version = "*", optional = true }
+pyyaml = { version = "*", optional = true }
+ruff = { version = "*", optional = true }
+sphinx = { version = "*", optional = true }
+myst-parser = { version = "*", optional = true }
+sphinx-autoapi = { version = "*", optional = true }
+sphinx-rtd-theme = { version = "*", optional = true }
```

Comment:
I think we also need `pre-commit` as a dev group and dev extra dependency.

Reply:
Good call, updated.